lib/src/application.dart

```diff
@@ -35,8 +35,8 @@ enum MetadataPersistenceMode {
 
 @immutable
 
-/// A class exposing configuration options for an [Application]
-class ApplicationConfiguration {
+/// A class exposing configuration options for an [App]
+class AppConfiguration {
```

Comment:
Looks okay to me.. one thing I have been wondering about.. Why did you drop the `application_configuration.dart` file and stuff everything in `application.dart`?

Reply:
its common in dart to use one file for multiple classes. No need to split these into separate files.